hera_cal/reflections.py

```diff
@@ -267,25 +239,445 @@ def write_reflections(self, output_calfits, input_calfits=None, overwrite=False,
+                       window='none', alpha=0.2, edgecut_low=0, edgecut_hi=0, verbose=True):
+        """
+        Subtract pcomp_model from data.
+
+        FFT pcomp_model to frequency space, divide by window, and subtract from data.
+        Inserts FFT of pcomp_model into self.pcomp_model_fft and 
+        residual of data with self.data_pmodel_resid. 
+
+        Note: The windowing parameters should be the *same* as those that were used
+        in constructing the dfft that pca_decomp operated on.
+
+        Args:
+            keys : list of tuples
+                List of baseline-pol DataContainer tuples to operate on.
+            data : datacontainer
+                Object to pull data from in forming data-model residual.
+                Default is self.data.
+            overwrite : bool
+                If True, overwrite output data if it exists.
+            ifft : bool
+                If True, use ifft to go from delay to freq axis
+            ifftshift : bool
+                If True, ifftshift delay axis before fft.
+            window : str
+                window function across freq to divide by after FFT.
+            alpha : float
+                if window is Tukey, this is its alpha parameter.
+            edgecut_low : int
+                Nbins to flag but not window at low-side of band.
+            edgecut_hi : int
+                Nbins to flag but not window at high-side of band.
+            verbose : bool
+                If True, report feedback to stdout.
+        """
+        # get keys
+        if keys is None:
+            keys = self.pcomp_model.keys()
+
+        if not hasattr(self, 'pcomp_model_fft'):
+            self.pcomp_model_fft = DataContainer({})
+        if not hasattr(self, 'data_pmodel_resid'):
+            self.data_pmodel_resid = DataContainer({})
+
+        # get data
+        if data is None:
+            data = self.data
+
+        # iterate over keys
+        for k in keys:
+            if k in self.pcomp_model_fft and not overwrite:
+                echo("{} in pcomp_model_fft and overwrite==False, skipping...".format(k), verbose=verbose)
+                continue
+
+            # get fft of model
+            model = self.pcomp_model[k]
+
+            # ifftshift
+            if ifftshift:
+                model = np.fft.ifftshift(model, axes=-1)
+
+            # ifft to get to data space
+            if ifft:
+                model_fft = np.fft.ifft(model, axis=-1)
+            else:
+                model_fft = np.fft.fft(model, axis=-1)
+
+            # divide by a window
+            win = dspec.gen_window(window, model_fft.shape[1], alpha=alpha,
+                                   edgecut_low=edgecut_low, edgecut_hi=edgecut_hi)
+            model_fft = np.true_divide(model_fft, win, where=~np.isclose(win, 0.0))
+
+            # subtract from data
+            self.pcomp_model_fft[k] = model_fft
```

Comment:
I'm getting concerned about the memory bloat of this code. By this point, you've stored at least half a dozen things as large as the data itself. Does it have to be this way?